sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/implementation/ChangeFeedState.java

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.implementation.changefeed.implementation;
+
+import com.azure.cosmos.implementation.JsonSerializable;
+import com.azure.cosmos.implementation.RxDocumentServiceRequest;
+import com.azure.cosmos.implementation.Utils;
+import com.azure.cosmos.implementation.feedranges.FeedRangeContinuation;
+import com.azure.cosmos.implementation.feedranges.FeedRangeInternal;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
+
+@JsonDeserialize(using = ChangeFeedStateDeserializer.class)
+public abstract class ChangeFeedState extends JsonSerializable {
+    ChangeFeedState() {
+    }
+
+    public abstract FeedRangeContinuation getContinuation();
+
+    public abstract ChangeFeedState setContinuation(FeedRangeContinuation continuation);
+
+    public abstract FeedRangeInternal getFeedRange();
+
+    public abstract ChangeFeedMode getMode();
+
+    public abstract ChangeFeedStartFromInternal getStartFromSettings();
+
+    public abstract String applyServerResponseContinuation(
+        String serverContinuationToken,
+        RxDocumentServiceRequest request);
+
+    public abstract String getContainerRid();
+
+    public static ChangeFeedState fromString(String base64EncodedJson) {
+        checkNotNull(base64EncodedJson, "Argument 'base64EncodedJson' must not be null");
+
+        String json = new String(
+            Base64.getUrlDecoder().decode(base64EncodedJson),
+            StandardCharsets.UTF_8);
+
+        final ObjectMapper mapper = Utils.getSimpleObjectMapper();
```

Comment:
Is getting the object mapper instance affected by other external settings/configs which could impact how we serialize/de-serialize the values? If true, we might need to pass our own configuration settings such that the Base64 encoded Json string is not impacted by such external changes.

Reply:
The Utils.getSimpleObjectMapper is used for the purpose of serializing other models - like PartitionKeyInternal, Offers etc. as well - so it wouldn't be possible to change the config without breaking many similar use cases. And no - the config is not passed down form the user etc. - completely implementation detail within Utils.java